src/ops/stash.rs

```diff
@@ -36,39 +41,88 @@ impl OpTrait for StashWorktree {
                 let input = state.prompt.state.value().to_string();
                 state.prompt.reset(term)?;
 
-                // 1. Stash index (stash@0: index, ...)
-                let mut cmd = Command::new("git");
-                cmd.args(["stash", "push", "--staged"]);
-                state.run_cmd(term, &[], cmd)?;
+                let need_to_stash_index = is_something_staged(&state.repo)?;
+
+                if need_to_stash_index {
+                    let mut cmd = Command::new("git");
+                    cmd.args(["stash", "push", "--staged"]);
+                    state.run_cmd(term, &[], cmd)?;
+                }
 
-                // 2. Stash everything else (stash@0: worktree, stash@1: index, ...)
                 let mut cmd = Command::new("git");
                 cmd.args(["stash", "push", "--include-untracked"]);
                 if !input.is_empty() {
                     cmd.args(["--message", &input]);
                 }
                 state.run_cmd(term, &[], cmd)?;
 
-                // 3. Pop stash with index (at stash@1)
-                let mut cmd = Command::new("git");
-                cmd.args(["stash", "pop", "1"]);
-                state.run_cmd(term, &[], cmd)?;
+                if need_to_stash_index {
+                    let mut cmd = Command::new("git");
+                    cmd.args(["stash", "pop", "1"]);
+                    state.run_cmd(term, &[], cmd)?;
+                }
             }
             Ok(())
         };
 
         Some(Rc::new(
-            move |state: &mut State, _term: &mut Term| -> Res<()> {
-                state.prompt.set(PromptData {
+            move |state: &mut State, term: &mut Term| -> Res<()> {
+                if is_working_tree_empty(&state.repo)? {
+                    state
+                        .current_cmd_log_entries
+                        .push(Arc::new(RwLock::new(CmdLogEntry::Error(
+                            "Cannot stash: working tree is empty".to_string(),
+                        ))));
+                    state.prompt.reset(term)?;
+                    return Ok(());
+                }
+                Ok(state.prompt.set(PromptData {
                     prompt_text: "Name of the stash:".into(),
                     update_fn: Rc::new(update_fn),
-                });
-                Ok(())
+                }))
```

Comment:
Clippy is complaining about this in the build i noticed.
The `ci.sh` script has some goodies i run locally myself.

Reply:
Yeah, I was running only `cargo test`, now I will run ci script. Very strange rule to be honest, this should be delegated to compiler :)
Done